File: semestr-8/Jezyk C#/alexander/Csharp/Archiwum/2021s/Id06IO1/Strategy/Strategy/Sportsmen.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Strategy
{
    public class Sportsmen
    {
        private string name;
        private StrategyRun srun;
        private StrategyJump sjump;
        private StrategySwim sswim;
        public Sportsmen(string name,StrategyRun srun,StrategyJump sjump,StrategySwim sswim)
        {
            this.name=name; 
            this.srun=srun;
            this.sjump=sjump;
            this.sswim=sswim;
        }
        public string Run() { return srun.execute(); }
        public string Swim() { return sswim.execute(); }
        public string Jump() { return sjump.execute(); }
        public string execute(SportsmenCommand cmd) { return name+": "+cmd.execute(this); }
    }
}

File: semestr-8/Programowanie obiektowe w Java/alexander/POJ/Archiwum/2023w/ID07IO1/StrategyCommandPattern/src/Sportsmen.java
public class Sportsmen
{
   private final String name;
   private final SportsmenSwimStrategy stSwim;

   public Sportsmen(String name,SportsmenSwimStrategy stSwim)
   {
      this.name=name;
      this.stSwim=stSwim;
   }

   public String Run() { return "Running"; }
   public String Jump() { return "Jumping"; }
   public String Swim() { return stSwim.Swim(); }
   @Override public String toString() { return name; }
}
